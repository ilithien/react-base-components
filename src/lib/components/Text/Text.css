.react-base__text {
  padding: 10px 5px 10px 10px;
  border-radius: 5px;
  border: 1px solid #ccc;
}

.react-base__text:focus {
  border: 1px solid #ddd;
}

.react-base__text-group {
  font-family: "Helvetica";
}

.react-base__text-group > label {
  display: block;
  margin-bottom: 5px;
  font-size: 14px;
  font-weight: 500;
  color: #bbb;
}
