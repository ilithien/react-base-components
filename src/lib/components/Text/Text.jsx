import React from "react";
import PropTypes from "prop-types";
import "./Text.css";

const Text = ({ id, name, placeholder, label, onChange }) => (
  <div className="react-base__text-group">
    <label htmlFor={id}>{label}</label>
    <input
      type="text"
      id={id}
      className="react-base__text"
      placeholder={placeholder}
      onChange={onChange}
      name={name}
    />
  </div>
);

Text.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
  label: PropTypes.string
};

export default Text;
