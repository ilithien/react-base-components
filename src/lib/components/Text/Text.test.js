import React from "react";
import renderer from "react-test-renderer";
import Text from "./Text";

describe("Testing Text component", () => {
  it("matches snapshot", () => {
    const snap = renderer.create(<Text />);

    const tree = snap.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
