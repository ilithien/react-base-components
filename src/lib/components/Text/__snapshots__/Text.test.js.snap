// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Text component matches snapshot 1`] = `
<div
  className="react-base__text-group"
>
  <label
    htmlFor={undefined}
  />
  <input
    className="react-base__text"
    id={undefined}
    name={undefined}
    onChange={undefined}
    placeholder={undefined}
    type="text"
  />
</div>
`;
